### DISCUSSION TITLE
Question on setup with parent copier template with child submodules that are also copier templates

### DISCUSSION THREAD

**Discussion by aorumbayev** [copier-org/copier#1191] [Q&A]
Given a parent repository (for sake of example, lets name it `fullstack`), that contains 2 git submodules which are self sufficient copier templates `backend` and `frontend`, is there any simple approach to define parent copier.yaml file which will:

1. Merge questions from both child templates (user does not get the overlapping questions asked twice)
2. Generate a parent template that will consist of 2 folders corresponding to instances of templates from submodule

I am aware of the copier recommendations on 1 template - 1 repo, hence looking for approaches that do not directly break that recommendation but still allow for code reuse without introducing a lot of duplication ( i deffo do not want to have rolled out instances of those child templates duplicated in the parent codebase). 

Any help or suggestions are appreciated! The only approach i figured so far is to use tasks, have an empty folder to be copied, overlapping questions from child copier.yamls merged together as a prerequisite task and then invoke scripts in that will in sequence copy the first child template, the second one and finalize the configuration.

Diagram below is roughly the estimate of what im trying to achieve:

```mermaid
graph TB
  CC["Copier Command"]
  CC -- "Runs on" --> P["Parent Repository (FullStack)"]
  P --> B["Backend (Git Submodule)"]
  P --> F["Frontend (Git Submodule)"]
  CC -- "Combine Questions" --> CQ["From Backend and Frontend copier.yaml"]
  CQ -- "Prompt User" --> Q["Generate Copier Answers"]
  Q -- "Initialize Backend Template" --> DB["Destination Folder/Backend"]
  Q -- "Initialize Frontend Template" --> DF["Destination Folder/Frontend"]
  linkStyle 0 stroke:#2ecd71,stroke-width:2px;
  linkStyle 1 stroke:#2ecd71,stroke-width:2px;
  linkStyle 2 stroke:#2ecd71,stroke-width:2px;
  linkStyle 3 stroke:#2ecd71,stroke-width:2px;
  linkStyle 4 stroke:#2ecd71,stroke-width:2px;
  linkStyle 5 stroke:#2ecd71,stroke-width:2px;
  linkStyle 6 stroke:#2ecd71,stroke-width:2px;
```

**Comment by yajo** [comment]
It all depends on the specific needs and characteristics of the project. 

Sometimes you can split the questionary between common and unique questions. Then you can include other questionaries in a single one.

Since all answers will be in context, you can use symlinks in `fullstack` to choose what to render from each of the children.

Or maybe you can share the most common parts of templates and use jinja include to DRY them.

Another option is to split `backend` and `frontend` into 2 templates that don't conflict with one another and let the user apply either or both to the same project.
